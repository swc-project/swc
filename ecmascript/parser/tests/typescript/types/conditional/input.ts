let x: number extends string ? boolean : null;
