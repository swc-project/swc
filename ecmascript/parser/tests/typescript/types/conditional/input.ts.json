{
  "type": "Module",
  "span": {
    "start": 0,
    "end": 46,
    "ctxt": 0,
    "loc": {
      "start": {
        "line": 1,
        "column": 0
      },
      "end": {
        "line": 1,
        "column": 46
      }
    }
  },
  "body": [
    {
      "type": "VariableDeclaration",
      "span": {
        "start": 0,
        "end": 46,
        "ctxt": 0,
        "loc": {
          "start": {
            "line": 1,
            "column": 0
          },
          "end": {
            "line": 1,
            "column": 46
          }
        }
      },
      "kind": "let",
      "declare": false,
      "declarations": [
        {
          "type": "VariableDeclarator",
          "span": {
            "start": 4,
            "end": 45,
            "ctxt": 0,
            "loc": {
              "start": {
                "line": 1,
                "column": 4
              },
              "end": {
                "line": 1,
                "column": 45
              }
            }
          },
          "id": {
            "type": "Identifier",
            "span": {
              "start": 4,
              "end": 5,
              "ctxt": 0,
              "loc": {
                "start": {
                  "line": 1,
                  "column": 4
                },
                "end": {
                  "line": 1,
                  "column": 5
                }
              }
            },
            "value": "x",
            "typeAnnotation": {
              "type": "TsTypeAnnotation",
              "span": {
                "start": 5,
                "end": 45,
                "ctxt": 0,
                "loc": {
                  "start": {
                    "line": 1,
                    "column": 5
                  },
                  "end": {
                    "line": 1,
                    "column": 45
                  }
                }
              },
              "typeAnnotation": {
                "type": "TsConditionalType",
                "span": {
                  "start": 7,
                  "end": 45,
                  "ctxt": 0,
                  "loc": {
                    "start": {
                      "line": 1,
                      "column": 7
                    },
                    "end": {
                      "line": 1,
                      "column": 45
                    }
                  }
                },
                "checkType": {
                  "type": "TsKeywordType",
                  "span": {
                    "start": 7,
                    "end": 13,
                    "ctxt": 0,
                    "loc": {
                      "start": {
                        "line": 1,
                        "column": 7
                      },
                      "end": {
                        "line": 1,
                        "column": 13
                      }
                    }
                  },
                  "kind": "number"
                },
                "extendsType": {
                  "type": "TsKeywordType",
                  "span": {
                    "start": 22,
                    "end": 28,
                    "ctxt": 0,
                    "loc": {
                      "start": {
                        "line": 1,
                        "column": 22
                      },
                      "end": {
                        "line": 1,
                        "column": 28
                      }
                    }
                  },
                  "kind": "string"
                },
                "trueType": {
                  "type": "TsKeywordType",
                  "span": {
                    "start": 31,
                    "end": 38,
                    "ctxt": 0,
                    "loc": {
                      "start": {
                        "line": 1,
                        "column": 31
                      },
                      "end": {
                        "line": 1,
                        "column": 38
                      }
                    }
                  },
                  "kind": "boolean"
                },
                "falseType": {
                  "type": "TsKeywordType",
                  "span": {
                    "start": 41,
                    "end": 45,
                    "ctxt": 0,
                    "loc": {
                      "start": {
                        "line": 1,
                        "column": 41
                      },
                      "end": {
                        "line": 1,
                        "column": 45
                      }
                    }
                  },
                  "kind": "null"
                }
              }
            },
            "optional": false
          },
          "definite": false
        }
      ]
    }
  ]
}
