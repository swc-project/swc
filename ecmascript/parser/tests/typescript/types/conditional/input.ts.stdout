Module {
    span: Span {
        lo: BytePos(
            0
        ),
        hi: BytePos(
            46
        ),
        ctxt: #0
    },
    body: [
        Stmt(
            Decl(
                Var(
                    VarDecl {
                        span: Span {
                            lo: BytePos(
                                0
                            ),
                            hi: BytePos(
                                46
                            ),
                            ctxt: #0
                        },
                        kind: "let",
                        declare: false,
                        decls: [
                            VarDeclarator {
                                span: Span {
                                    lo: BytePos(
                                        4
                                    ),
                                    hi: BytePos(
                                        45
                                    ),
                                    ctxt: #0
                                },
                                name: Ident(
                                    Ident {
                                        sym: x,
                                        span: Span {
                                            lo: BytePos(
                                                4
                                            ),
                                            hi: BytePos(
                                                5
                                            ),
                                            ctxt: #0
                                        },
                                        type_ann: Some(
                                            TsTypeAnn {
                                                span: Span {
                                                    lo: BytePos(
                                                        5
                                                    ),
                                                    hi: BytePos(
                                                        45
                                                    ),
                                                    ctxt: #0
                                                },
                                                type_ann: TsConditionalType(
                                                    TsConditionalType {
                                                        span: Span {
                                                            lo: BytePos(
                                                                7
                                                            ),
                                                            hi: BytePos(
                                                                45
                                                            ),
                                                            ctxt: #0
                                                        },
                                                        check_type: TsKeywordType(
                                                            TsKeywordType {
                                                                span: Span {
                                                                    lo: BytePos(
                                                                        7
                                                                    ),
                                                                    hi: BytePos(
                                                                        13
                                                                    ),
                                                                    ctxt: #0
                                                                },
                                                                kind: TsNumberKeyword
                                                            }
                                                        ),
                                                        extends_type: TsKeywordType(
                                                            TsKeywordType {
                                                                span: Span {
                                                                    lo: BytePos(
                                                                        22
                                                                    ),
                                                                    hi: BytePos(
                                                                        28
                                                                    ),
                                                                    ctxt: #0
                                                                },
                                                                kind: TsStringKeyword
                                                            }
                                                        ),
                                                        true_type: TsKeywordType(
                                                            TsKeywordType {
                                                                span: Span {
                                                                    lo: BytePos(
                                                                        31
                                                                    ),
                                                                    hi: BytePos(
                                                                        38
                                                                    ),
                                                                    ctxt: #0
                                                                },
                                                                kind: TsBooleanKeyword
                                                            }
                                                        ),
                                                        false_type: TsKeywordType(
                                                            TsKeywordType {
                                                                span: Span {
                                                                    lo: BytePos(
                                                                        41
                                                                    ),
                                                                    hi: BytePos(
                                                                        45
                                                                    ),
                                                                    ctxt: #0
                                                                },
                                                                kind: TsNullKeyword
                                                            }
                                                        )
                                                    }
                                                )
                                            }
                                        ),
                                        optional: false
                                    }
                                ),
                                init: None,
                                definite: false
                            }
                        ]
                    }
                )
            )
        )
    ],
    shebang: None
}
